### PR TITLE
fix: reset-password code coverage

### PR DESCRIPTION
**Pull request by ahtesham-quraish**
Description
Improve the code coverage of reset-password component

How Has This Been Tested?
Its been tested locally 

Related to: https://2u-internal.atlassian.net/browse/VAN-1369

### CODE REVIEW

File: src/reset-password/tests/ResetPasswordPage.test.jsx [hunk 1]
```diff
@@ -157,6 +157,55 @@ describe('ResetPasswordPage', () => {
 
   // ******** miscellaneous tests ********
 
+  it('check password value on blur and enable show password field', () => {
+    const resetPasswordPage = mount(reduxWrapper(<IntlResetPasswordPage {...props} />));
+    resetPasswordPage.find('input#newPassword').simulate('change', { target: { value: 'aziz156@', name: 'newPassword' } });
+    resetPasswordPage.find('input#newPassword').simulate('blur', { target: { value: 'aziz156@', name: 'newPassword' } });
+    expect(resetPasswordPage.find('input#newPassword').props().value).toEqual('aziz156@');
+  });
```
Comment (mubbsharanwar) [inline src/reset-password/tests/ResetPasswordPage.test.jsx:165]: Could you elaborate this test? What we are doing in this test?

File: src/reset-password/tests/ResetPasswordPage.test.jsx [hunk 2]
```diff
@@ -157,6 +157,55 @@ describe('ResetPasswordPage', () => {
...
+        params: { token: '1c-bmjdkc-5e60e084cf8113048ca7' },
+      },
+    };
+    mount(reduxWrapper(<IntlResetPasswordPage {...props} />));
+    expect(store.dispatch).toHaveBeenCalledWith(validateToken(props.match.params.token));
+  });
+  it('should redirect the user to Reset password email screen ', async () => {
+    props = {
+      status:
+      'password-reset-error',
```
Comment (mubbsharanwar) [inline src/reset-password/tests/ResetPasswordPage.test.jsx:195]: Remove this hardcoded const and import this from the constant file.

File: src/reset-password/tests/ResetPasswordPage.test.jsx [hunk 3]
```diff
@@ -157,6 +157,55 @@ describe('ResetPasswordPage', () => {
...
+    };
+    const resetPasswordPage = mount(reduxWrapper(<IntlResetPasswordPage {...props} />));
+    expect(resetPasswordPage.find('input#email')).toBeTruthy();
+  });
+  it('should redirect the user to root url of the application ', async () => {
```
Comment (mubbsharanwar) [inline src/reset-password/tests/ResetPasswordPage.test.jsx:200]: In the success case, the user would redirect to the login page. As you are passing the success status in props.

File: src/reset-password/tests/ResetPasswordPage.test.jsx [hunk 4]
```diff
@@ -157,6 +157,55 @@ describe('ResetPasswordPage', () => {
...
+    props = {
+      status: 'success',
```
Comment (mubbsharanwar) [inline src/reset-password/tests/ResetPasswordPage.test.jsx:202]: Import this const from constant file.

File: src/reset-password/tests/ResetPasswordPage.test.jsx [hunk 5]
```diff
@@ -157,6 +157,55 @@ describe('ResetPasswordPage', () => {
...
+    };
+    const resetPasswordPage = mount(reduxWrapper(<IntlResetPasswordPage {...props} />));
+
+    expect(resetPasswordPage.find('input#email')).toBeTruthy();
```
Comment (mubbsharanwar) [inline src/reset-password/tests/ResetPasswordPage.test.jsx:206]: Also in expect there should be `LoginPage` component

File: src/reset-password/tests/ResetPasswordPage.test.jsx [hunk 6]
```diff
@@ -157,6 +157,55 @@ describe('ResetPasswordPage', () => {
...
+      status:
+      'token-pending',
+      match: {
+        params: { token: '1c-bmjdkc-5e60e084cf8113048ca7' },
+      },
+    };
+    mount(reduxWrapper(<IntlResetPasswordPage {...props} />));
+    expect(store.dispatch).toHaveBeenCalledWith(validateToken(props.match.params.token));
+  });
+  it('should redirect the user to Reset password email screen ', async () => {
```
Comment (mubbsharanwar) [inline src/reset-password/tests/ResetPasswordPage.test.jsx:192]: Do you mean Forgot password page?
If yes then you need to check `ForgotPasswordPage` instead `ResetPasswordPage` in expect statement.

**Review by mubbsharanwar — CHANGES_REQUESTED** [review]

File: src/reset-password/tests/ResetPasswordPage.test.jsx (hunk 2 above)
Comment (ahtesham-quraish) [reply src/reset-password/tests/ResetPasswordPage.test.jsx:195]: addressed

File: src/reset-password/tests/ResetPasswordPage.test.jsx (hunk 1 above)
Comment (ahtesham-quraish) [reply src/reset-password/tests/ResetPasswordPage.test.jsx:165]: addressed

File: src/reset-password/tests/ResetPasswordPage.test.jsx (hunk 4 above)
Comment (ahtesham-quraish) [reply src/reset-password/tests/ResetPasswordPage.test.jsx:202]: aadressed

File: src/reset-password/tests/ResetPasswordPage.test.jsx (hunk 6 above)
Comment (ahtesham-quraish) [reply src/reset-password/tests/ResetPasswordPage.test.jsx:194]: addressed

File: src/reset-password/tests/ResetPasswordPage.test.jsx (hunk 5 above)
Comment (ahtesham-quraish) [reply src/reset-password/tests/ResetPasswordPage.test.jsx]: addressed

File: src/reset-password/tests/ResetPasswordPage.test.jsx (hunk 3 above)
Comment (ahtesham-quraish) [reply src/reset-password/tests/ResetPasswordPage.test.jsx:207]: addressed

**Pull request merged**